test/clj/cider/nrepl/middleware/apropos_test.clj

```diff
@@ -39,3 +49,61 @@
       (is (= (take 20 (:doc x))
              (take 20 (:doc y)))
           "The abbreviated docstring should be the start of the full docstring."))))
+
+(use-fixtures :each session/session-fixture)
+(deftest integration-test
+  (testing "Apropos op, typical case"
+    (let [response (session/message {:op "apropos" :query "handle-apropos"})
+          match    (get-in response [:apropos-matches 0])]
+      (is (= (:status response) #{"done"}))
+      (is (= (:type match) "function"))
+      (is (= (:name match) "cider.nrepl.middleware.apropos/handle-apropos"))
+      (is (.startsWith (:doc match) "Return a sequence of vars"))))
+
+  (testing "Apropos op, but specialized cases (invoked with prefix argument)"
```

Comment:
I'd break this into several `testing` blocks - in a way the string passed to testing serves as the description you provide by using inline comments.